### PR TITLE
update RN50 script with Intel GPU support

### PR DESCRIPTION
**Pull request by ashokei**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/GOVERNANCE.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 1]
```diff
@@ -205,77 +324,106 @@ def avg(self):
...
 
     # Set up standard ResNet-50 model.
-    model = models.resnet50()
+    print("=> creating model '{}'".format(args.arch))
+    model = models.__dict__[args.arch]()
 
     # By default, Adasum doesn't need scaling up learning rate.
     # For sum/average with gradient Accumulation: scale learning rate by batches_per_allreduce
-    lr_scaler = args.batches_per_allreduce * hvd.size() if not args.use_adasum else 1
+    lr_scaler = args.batches_per_allreduce * hvd.size(
```
Comment (zhangxiaoli73) [inline examples/pytorch/pytorch_imagenet_resnet50.py:390]: Please keep same format as stock if you don't change the code.

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 2]
```diff
@@ -1,59 +1,132 @@
...
+
+model_names = sorted(name for name in models.__dict__
+                     if name.islower() and not name.startswith("__")
+                     and callable(models.__dict__[name]))
 
 # Training settings
-parser = argparse.ArgumentParser(description='PyTorch ImageNet Example',
-                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-parser.add_argument('--train-dir', default=os.path.expanduser('~/imagenet/train'),
+parser = argparse.ArgumentParser(
```
Comment (zhangxiaoli73) [inline examples/pytorch/pytorch_imagenet_resnet50.py:19]: Please keep same format as stock if you don't change the code.

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 3]
```diff
@@ -1,59 +1,132 @@
...
+                    help='using bfp16')
+parser.add_argument('--warm', type=int, default=10, help='warmup for fps')
+parser.add_argument('--iter', type=int, default=None, help='limit iterations')
+parser.add_argument('--dummy',
+                    action="store_true",
+                    help='use dummy data for '
+                    'benchmark training or val')
+
+
+def _time(args):
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:124]: This is very controversial.  I am entirely against adding artificial overheads.

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 4]
```diff
@@ -64,13 +137,26 @@ def train(epoch):
 
     with tqdm(total=len(train_loader),
               desc='Train Epoch     #{}'.format(epoch + 1),
-              disable=not verbose) as t:
+              disable=not verbose,
+              ascii=True) as t:
+        t1 = _time(args)
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:142]: We should not add these.

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 5]
```diff
@@ -84,9 +170,13 @@ def train(epoch):
                 loss.backward()
             # Gradient is applied across all ranks
             optimizer.step()
-            t.set_postfix({'loss': train_loss.avg.item(),
-                           'accuracy': 100. * train_accuracy.avg.item()})
+            t.set_postfix({
+                'loss': train_loss.avg.item(),
+                'accuracy': 100. * train_accuracy.avg.item(),
+                'fps': imgs / (_time(args) - t1)
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:176]: We  should not change this.

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 6]
```diff
@@ -100,17 +190,30 @@ def validate(epoch):
 
     with tqdm(total=len(val_loader),
               desc='Validate Epoch  #{}'.format(epoch + 1),
-              disable=not verbose) as t:
+              disable=not verbose,
+              ascii=True) as t:
+        t1 = _time(args)
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:195]: We should not add this here,

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 7]
```diff
@@ -100,17 +190,30 @@ def validate(epoch):
...
                 output = model(data)
 
                 val_loss.update(F.cross_entropy(output, target))
                 val_accuracy.update(accuracy(output, target))
-                t.set_postfix({'loss': val_loss.avg.item(),
-                               'accuracy': 100. * val_accuracy.avg.item()})
+                t.set_postfix({
+                    'loss': val_loss.avg.item(),
+                    'accuracy': 100. * val_accuracy.avg.item(),
+                    'fps': imgs / (_time(args) - t1)
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:215]: We should not change this .

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 8]
```diff
@@ -1,59 +1,132 @@
...
+                    help='use xpu')
+parser.add_argument('--nohorovod',
+                    action='store_true',
+                    default=False,
+                    help='no horovod')
+parser.add_argument('--nhwc',
+                    action='store_true',
+                    default=False,
+                    help='using channel last tensor layout')
+parser.add_argument('--bf16',
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:112]: Do we need bf16 ?

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 9]
```diff
@@ -1,59 +1,132 @@
...
+                    action='store_true',
+                    default=False,
+                    help='using bfp16')
+parser.add_argument('--warm', type=int, default=10, help='warmup for fps')
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:116]: Should not add this.

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 10]
```diff
@@ -1,59 +1,132 @@
...
+parser.add_argument('--iter', type=int, default=None, help='limit iterations')
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:117]: Should not add this.

File: examples/pytorch/pytorch_imagenet_resnet50.py [hunk 11]
```diff
@@ -1,59 +1,132 @@
...
+parser.add_argument('--dummy',
```
Comment (githubsgi) [inline examples/pytorch/pytorch_imagenet_resnet50.py:118]: Should not add this.

**Review by githubsgi — COMMENTED** [review]
See my comments inline.

**Pull request closed without merging**